pkg/quota/image/imagestream_evaluator.go

```diff
@@ -25,50 +25,53 @@ const (
 // NewImageStreamEvaluator computes resource usage of ImageStreams. Instantiating this is necessary for
 // resource quota admission controller to properly work on image stream related objects.
 func NewImageStreamEvaluator(osClient osclient.Interface) kquota.Evaluator {
-	computeResources := []kapi.ResourceName{
-		imageapi.ResourceImages,
-	}
-
-	matchesScopeFunc := func(kapi.ResourceQuotaScope, runtime.Object) bool { return true }
-	getFuncByNamespace := func(namespace, name string) (runtime.Object, error) {
-		return osClient.ImageStreams(namespace).Get(name)
-	}
 	listFuncByNamespace := func(namespace string, options kapi.ListOptions) (runtime.Object, error) {
 		return osClient.ImageStreams(namespace).List(options)
 	}
 
+	evaluator := NewImageStreamAdmissionEvaluator(osClient)
+	genericEvaluator := evaluator.(*generic.GenericEvaluator)
+	genericEvaluator.Name = imageStreamEvaluatorName
+	genericEvaluator.InternalOperationResources = nil
+	genericEvaluator.ListFuncByNamespace = listFuncByNamespace
+
 	return quotautil.NewSharedContextEvaluator(
-		imageStreamEvaluatorName,
-		kapi.Kind("ImageStream"),
-		nil,
-		computeResources,
-		matchesScopeFunc,
-		getFuncByNamespace,
-		listFuncByNamespace,
-		imageStreamConstraintsFunc,
+		genericEvaluator,
 		makeImageStreamUsageComputerFactory(osClient))
 }
 
 // NewImageStreamAdmissionEvaluator computes resource usage of ImageStreams in the context of admission
 // plugin.
 func NewImageStreamAdmissionEvaluator(osClient osclient.Interface) kquota.Evaluator {
-	evaluator := NewImageStreamEvaluator(osClient)
-	isEval := evaluator.(*quotautil.SharedContextEvaluator)
-	isEval.Name = imageStreamAdmissionEvaluatorName
-	isEval.InternalOperationResources = map[admission.Operation][]kapi.ResourceName{
-		admission.Create: isEval.MatchedResourceNames,
-		admission.Update: isEval.MatchedResourceNames,
+	computeResources := []kapi.ResourceName{
```

Comment:
Got to here and I need a rest.  Setting some fields and not others (`ListFuncByNamespace`?) isn't something my mind can still reason about.  Long pull.